plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationCompassEngine.kt

```diff
@@ -0,0 +1,321 @@
+        // Remap the axes as if the device screen was upside down and facing back.
+        when (windowManager.defaultDisplay.rotation) {
+          Surface.ROTATION_90 -> {
+            worldAxisForDeviceAxisX1 = SensorManager.AXIS_MINUS_Z
+            worldAxisForDeviceAxisY1 = SensorManager.AXIS_MINUS_X
+          }
+          Surface.ROTATION_180 -> {
+            worldAxisForDeviceAxisX1 = SensorManager.AXIS_MINUS_X
+            worldAxisForDeviceAxisY1 = SensorManager.AXIS_Z
+          }
+          Surface.ROTATION_270 -> {
+            worldAxisForDeviceAxisX1 = SensorManager.AXIS_Z
+            worldAxisForDeviceAxisY1 = SensorManager.AXIS_X
+          }
+          Surface.ROTATION_0 -> {
+            worldAxisForDeviceAxisX1 = SensorManager.AXIS_X
+            worldAxisForDeviceAxisY1 = SensorManager.AXIS_MINUS_Z
+          }
+          else -> {
+            worldAxisForDeviceAxisX1 = SensorManager.AXIS_X
+            worldAxisForDeviceAxisY1 = SensorManager.AXIS_MINUS_Z
+          }
+        }
+      }
+      Math.abs(orientation[2]) > Math.PI / 2 -> {
+        // The roll is less than -90 degrees, or is larger than 90 degrees.
+        // Remap the axes as if the device screen was face down.
+        when (windowManager.defaultDisplay.rotation) {
+          Surface.ROTATION_90 -> {
+            worldAxisForDeviceAxisX1 = SensorManager.AXIS_MINUS_Y
+            worldAxisForDeviceAxisY1 = SensorManager.AXIS_MINUS_X
+          }
+          Surface.ROTATION_180 -> {
+            worldAxisForDeviceAxisX1 = SensorManager.AXIS_MINUS_X
+            worldAxisForDeviceAxisY1 = SensorManager.AXIS_Y
+          }
+          Surface.ROTATION_270 -> {
+            worldAxisForDeviceAxisX1 = SensorManager.AXIS_Y
+            worldAxisForDeviceAxisY1 = SensorManager.AXIS_X
+          }
+          Surface.ROTATION_0 -> {
+            worldAxisForDeviceAxisX1 = SensorManager.AXIS_X
+            worldAxisForDeviceAxisY1 = SensorManager.AXIS_MINUS_Y
+          }
+          else -> {
+            worldAxisForDeviceAxisX1 = SensorManager.AXIS_X
+            worldAxisForDeviceAxisY1 = SensorManager.AXIS_MINUS_Y
+          }
+        }
+      }
+    }
+    return Pair(worldAxisForDeviceAxisX1, worldAxisForDeviceAxisY1)
+  }
+
+  private fun getWorldAxisFromRotation(): Pair<Int, Int> {
+    var worldAxisForDeviceAxisX: Int
+    var worldAxisForDeviceAxisY: Int
+    when (windowManager.defaultDisplay.rotation) {
+      Surface.ROTATION_90 -> {
+        worldAxisForDeviceAxisX = SensorManager.AXIS_Y
+        worldAxisForDeviceAxisY = SensorManager.AXIS_MINUS_X
+      }
+      Surface.ROTATION_180 -> {
+        worldAxisForDeviceAxisX = SensorManager.AXIS_MINUS_X
+        worldAxisForDeviceAxisY = SensorManager.AXIS_MINUS_Y
+      }
+      Surface.ROTATION_270 -> {
+        worldAxisForDeviceAxisX = SensorManager.AXIS_MINUS_Y
+        worldAxisForDeviceAxisY = SensorManager.AXIS_X
+      }
+      else -> {
+        worldAxisForDeviceAxisX = SensorManager.AXIS_X
```

Comment:
what about `Rotation_0`? if it's merged with `else` block maybe do the same for other `when` blocks?

Reply:
The else block is actually `Rotation_0`, we don't have any other conditions.

Reply:
I mean you've got `Surface.ROTATION_0 ->` and `else` block in another checks also

```
Surface.ROTATION_0 -> {
            worldAxisForDeviceAxisX1 = SensorManager.AXIS_X
            worldAxisForDeviceAxisY1 = SensorManager.AXIS_MINUS_Y
          }
          else -> {
            worldAxisForDeviceAxisX1 = SensorManager.AXIS_X
            worldAxisForDeviceAxisY1 = SensorManager.AXIS_MINUS_Y
          }
```

```
 Surface.ROTATION_0 -> {
            worldAxisForDeviceAxisX1 = SensorManager.AXIS_X
            worldAxisForDeviceAxisY1 = SensorManager.AXIS_MINUS_Z
          }
          else -> {
            worldAxisForDeviceAxisX1 = SensorManager.AXIS_X
            worldAxisForDeviceAxisY1 = SensorManager.AXIS_MINUS_Z
          }
```

Reply:
Oh right, nice catch